src/utils/changeElementVOVisibility.js

```diff
@@ -27,43 +59,46 @@ function iterateOverSiblings(el, callback) {
   }
 }
 
-const PREFIX = 'data-original-';
-const prop = 'aria-hidden';
-const prefixedProperty = PREFIX + prop;
-
 /**
  * Hides an element from VO
  * @param {HTMLElement} element
  */
 const hideElement = (element) => {
-  let originalValue = element.getAttribute(prefixedProperty);
-  if (!originalValue) {
-    // store the prop temporarily, to retrieve later.
-    originalValue = element.getAttribute(prop) || '';
-    element.setAttribute(prefixedProperty, originalValue);
+  // set original value for prefixed properties and tabindex
+  // store the prop temporarily, to retrieve later.
+  setOriginalValue(element, ARIA);
+  setOriginalValue(element, TABINDEX);
+
+  // hide the component from VO
+  element.setAttribute(ARIA, 'true');
+
+  // hide the component from tabbing
+  element.setAttribute(TABINDEX, '-1');
+  // make sure element's tabbable children are hidden as well
+  const tabbables = TabManager.getTabbableElements(element);
+  let i = tabbables.length - 1;
+  while (i >= 0) {
+    setOriginalValue(tabbables[i], TABINDEX);
+    tabbables[i].setAttribute(TABINDEX, '-1');
+    i -= 1;
   }
-  // hide the component
-  element.setAttribute(prop, 'true');
 };
 
 /**
  * Show an element
  * @param {HTMLElement} element
  */
 const showElement = (element) => {
-  // get the cached property
-  const originalValue = element.getAttribute(prefixedProperty);
-  if (typeof originalValue === 'string') {
-    // if there is a value, set it back.
-    if (originalValue.length) {
-      element.setAttribute(prop, originalValue);
-    } else {
-      // otherwise remove the attribute entirely.
-      element.removeAttribute(prop);
-    }
+  retrieveOriginalValue(element, ARIA);
+  retrieveOriginalValue(element, TABINDEX);
+
+  // make sure element's tabbable children are restored as well
+  const tabbables = element.querySelectorAll(`[${PREFIX + TABINDEX}]`);
```

Comment:
`TabManager.getTabbableElements` would not work here now, would it? 🤔 Because they are no longer tabbable.

Reply:
Yup. If we use `getTabbableElements`, the only ones that would get restored would be the ones that are focusable by nature.